src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+        }
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core PUT "$local:RelPath" -Body $local:tagObj
+    } else {
+        throw "The tag with id $TagId could not be found."
+    }
+}
+
+
+<#
+.SYNOPSIS
+Remove a tag from Safeguard via the Web API.
+
+.DESCRIPTION
+Remove a tag from Safeguard. 
+If the tag is assigned to an asset or an account, a SafeguardMethodException is thrown.
+Make sure it is not in use before you remove it.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID  or a string containing the name of the asset partition to delete a tag form.
+
+.PARAMETER AssetPartitionId
+An integer containing the asset partition ID to delete a tag form.
+(If specified, this will override the AssetPartition parameter)
+
+.PARAMETER TagToDelete
+An integer containing the ID of the asset to remove or a string containing the name of the tag.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Remove-SafeguardTag -AccessToken $token -Appliance 10.5.32.54 -Insecure 5
+
+.EXAMPLE
+Remove-SafeguardTag prod
+#>
+function Remove-SafeguardTag {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$TagToDelete
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    # use the helper function Resolve-AssetPartitionId to get the asset partition id.
+    $AssetPartitionId = Resolve-AssetPartitionId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId
+    $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags"    
+    $local:TagId = (Resolve-SafeguardTagId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId $TagToDelete)
+
+    Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core DELETE "$local:RelPath/$($local:TagId)"
+}
+
+
+<#
+.SYNOPSIS
+Test an AssetTaggingRule on a tag.
+
+.DESCRIPTION
+Test what an AssetTaggingRule would do. 
+This can be used to verify which assets would be assigned this the tag if the rule is set.
```

Comment:
"...would be assigned this the tag if..." Note the extra "the" after "this". Could be "...would be assigned this tag if..."